<meta charset=utf8>
<span>
help: https://github.com/igvteam/igv.js/wiki 
| 
test5_8C: multiple bams, hg38 <b class="red">HeLa Fucci(GSE104736) & MDA-MB-231</b>, add using js; {colorBy: "sample"}; {hg38: GAPDH, ACTB, };
</span>
<style>
.red{color:red;}
</style>

<script src="js/igv.min.js"></script>
<div id="igv-div"></div>

<script>
// ********************* resources *********************

// refer in local
var hg38_local={
	"id": "hg38",
	"name": "Human (GRCh38/hg38)",
	"fastaURL": "ref/hg38/hg38.fa", //?someRandomSeed=0
	"indexURL": "ref/hg38/hg38.fa.fai",
	"cytobandURL": "ref/hg38/cytoBandIdeo.txt.gz",
	"aliasURL": "ref/hg38/hg38_alias.tab",
	
	"tracks": [
	  {
		"name": "Refseq Genes",
		"format": "refgene",
		"id": "hg38_genes",
		"url": "ref/hg38/ncbiRefSeq.txt.gz",
		"indexURL": "ref/hg38/ncbiRefSeq.txt.gz.tbi",
		"visibilityWindow": -1,
		"supportsWholeGenome": false,
		"removable": false,
		"order": 1000000,
		height:250,   //height of ref track
		"infoURL": "https://www.ncbi.nlm.nih.gov/gene/?term=$$"  //links to ncbi gene
	  }
	],
	"chromosomeOrder": "chr1, chr2, chr3, chr4, chr5, chr6, chr7, chr8, chr9, chr10, chr11, chr12, chr13, chr14, chr15, chr16, chr17, chr18, chr19, chr20, chr21, chr22, chrX, chrY"
}

//1. settings
var options = {
    //genome: "hg38",
    reference: hg38_local,
    locus: 'chr1:246,766,264-246,771,021', //"chr10:78,039,965-78,041,277",
    tracks: []
};



// add bed file of pA sites
var oBed={
	"name": "polyA DB2",
	"type": "annotation",
	"format": "bed",
	"url": "http://z.biomooc.com:8892/list/polyA_DB/hg38.PolyADB2-liftover.bed",
	"color": "red",
	height:50, //total height: default 300
}
options.tracks.push(oBed);




var colors=[ //"#A80000", 
    "#eeeeee",
	"#DC143C","#0000FF","#20B2AA","#FFA500","#9370DB","#98FB98","#F08080","#1E90FF","#7CFC00","#FFFF00",
	"#808000","#FF00FF","#FA8072","#7B68EE","#9400D3","#800080","#A0522D","#D2B48C","#D2691E","#87CEEB",
	"#40E0D0","#5F9EA0"]

//add whole bam
var sampleColors = ["blue", "red", "green"]
var phases = ["G1", "S", "G2"]

// cell cycle: HeLa
for(var j=0; j<phases.length; j++){
	var phase = phases[j];
	for(var i=1; i<4; i++){
		//if(i>1){ continue; }

        var key = phase+"_rep"+i;
        var bam_path = "http://z.biomooc.com:8892/list/map/"+key+"_Aligned.sortedByCoord.out.bam";
		var oTrack={
			"name": "HeLa_"+key,
			"url":     bam_path ,
			"indexURL": bam_path + ".bai",
			"format": "bam",
			"type": "alignment",
			"color": sampleColors[j],
			//colorBy: "strand", // for alignment read
			showAlignments: !true,

			height:30, //150, //total height: default 300
			coverageTrackHeight:30, //coverage height: default 50
			alignmentRowHeight:5, //height of reads, default 14 per reads;
		}
        console.log("loading bam: ", key)
		options.tracks.push(oTrack);
	}
}

// cell cycle: U2OS
for(var j=0; j<phases.length; j++){
	var phase = phases[j];
	for(var i=1; i<3; i++){
		//if(i>1){ continue; }

        var key = "U2OS_"+phase+"_rep"+i;
        var bam_path = "http://z.biomooc.com:8892/list/map/"+key+"_Aligned.sortedByCoord.out.bam";
		var oTrack={
			"name": key,
			"url":     bam_path ,
			"indexURL": bam_path + ".bai",
			"format": "bam",
			"type": "alignment",
			"color": sampleColors[j],
			//colorBy: "strand", // for alignment read
			showAlignments: !true,

			height:30, //150, //total height: default 300
			coverageTrackHeight:30, //coverage height: default 50
			alignmentRowHeight:5, //height of reads, default 14 per reads;
		}
        console.log("loading bam: ", key)
		options.tracks.push(oTrack);
	}
}




//MDA-MB-231 v2
/* https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE216497
MDA-MB-231 cells were harvested at 0, 3, 4.5, 6, 7, 9, 10, 11.5, 13, and 16 hr after G1/S phase release. 
For M phase arrest, when MDA-MB-231 cells were released from G1/S phase, 1 µg/ml nocodazole was added, and after 10 hr, 
they were supplemented with 5 µg/ml actinomycin D.
*/
var cellname="MDA-MB-231"
var SRR = [
    "SRR22029273","SRR22029272","SRR22029271","SRR22029270", "SRR22029269", "SRR22029268","SRR22029267","SRR22029266", "SRR22029265","SRR22029264",   
    "SRR22029263", 
    "SRR22029259", "SRR22029260", "SRR22029261", "SRR22029262", "SRR22029258", 
      ]
var times = ["S0", "S3", "S4.5", "S6", "S7", "S9", "S10", "S11.5", "S13", "S16", 
    "ctrl", 
    "ActD0", "ActD1.5", "ActD3", "ActD4.5", "ActD6"]
for(var j=0; j<SRR.length; j++){
	var phase = SRR[j];
	//if( (j%3) >=2 ){continue;}
	//for(var i=1; i<3; i++){
        var key = phase;
        var bam_path = "http://z.biomooc.com:8892/list/"+cellname+"/"+key+"_Aligned.sortedByCoord.out.bam";
		var oTrack={
			"name": cellname + "_"+ times[j],
			"url":     bam_path ,
			"indexURL": bam_path + ".bai",
			"format": "bam",
			"type": "alignment",
			"color": colors[ j<10?3:(j==10?4:5)], //"deeppink",
			//colorBy: "strand", // for alignment read
			showAlignments: !true,

			height:30, //150, //total height: default 300
			coverageTrackHeight:30, //coverage height: default 50
			alignmentRowHeight:0, //height of reads, default 14 per reads;
		}
        console.log("loading bam: ", key)
		options.tracks.push(oTrack);
	//}
}



// add bed file of pA sites
var oBed={
		"name": "polyA DB2",
		"type": "annotation",
		"format": "bed",
		"url": "http://z.biomooc.com:8892/list/polyA_DB/hg38.PolyADB2-liftover.bed",
		"color": "red",
		height:50, //total height: default 300
	}
options.tracks.push(oBed);
var oBed={
		"name": "polyA DB3",
		"type": "annotation",
		"format": "bed",
		"url": "http://z.biomooc.com:8892/list/polyA_DB/hg38.PolyADB3-liftover.bed",
		"color": "black",
		height:50, //total height: default 300
	}
options.tracks.push(oBed);







//2. run IGV
var browser;

var igvDiv = document.getElementById("igv-div");
igv.createBrowser(igvDiv, options)
    .then(function (browser2) {
        console.log("Created IGV browser");	
        browser=browser2;
    })
// view a given range on genome, exec on F12 console
// browser.search("chr10:96,750,095-96,751,424")

/*
get the current locus(or range or view) 
> browser.toJSON().locus
'chr10:132,549,209-132,570,201'

*/
</script>
